lib/providers/acr.bash

```
@@ -0,0 +1,86 @@
+#!/bin/bash
+
+# Azure Container Registry provider for Docker push plugin
+
+setup_acr_environment() {
+  local registry_url_raw="${BUILDKITE_PLUGIN_DOCKER_IMAGE_PUSH_ACR_REGISTRY_URL:-}"
+  local acr_login="${BUILDKITE_PLUGIN_DOCKER_IMAGE_PUSH_ACR_ACR_LOGIN:-false}"
+
+  # Validate registry URL is always required
+  if [[ -z "$registry_url_raw" ]]; then
+    log_error "ACR registry URL is required"
+    log_info "Set it via the 'registry-url' parameter in acr configuration"
+    exit 1
+  fi
+
+  # Process environment variable reference for registry URL
+  local registry_url
+  registry_url=$(expand_env_var "$registry_url_raw" "registry-url")
+
+  # Clean up registry URL (remove protocol if present)
+  registry_url="${registry_url#https://}"
+  registry_url="${registry_url#http://}"
+
+  export DOCKER_PUSH_ACR_REGISTRY_URL="$registry_url"
+
+  log_info "ACR registry URL: $registry_url"
+
+  if [[ "$acr_login" =~ ^(true|1)$ ]]; then
+    # Use Azure CLI authentication
+    if ! command_exists az; then
+      log_error "Azure CLI (az) is required when acr-login is enabled"
+      log_info "Install the Azure CLI or use service principal authentication instead"
+      exit 1
+    fi
```

Uses `command_exists` to check if `az` available, if user sets `acr-login: true` and `az` is not available we error instead of silently falling back to using other creds